lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2244,20 +2242,20 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
     diags.diagnose(match.Witness, diag::protocol_witness_not_objc);
     break;
   // SWIFT_ENABLE_TENSORFLOW
-  case MatchKind::DifferentiableConflict:
-    std::string diffAttrReq;
-    {
+  case MatchKind::DifferentiableConflict: {
+    for (auto *da : req->getAttrs()
+             .getAttributes<DifferentiableAttr, /*allowInvalid*/ true>()) {
+      assert(da);
+      std::string diffAttrReq;
       llvm::raw_string_ostream stream(diffAttrReq);
-      // TODO(TF-285): Handle multiple `@differentiable` attributes on protocol
-      // requirements. Only missing attributes should be diagnosed.
-      req->getAttrs().getAttribute<DifferentiableAttr>()->print(stream, req);
-      diffAttrReq = StringRef(stream.str()).trim();
+      da->print(stream);
```

Comment:
Is there a reason that you don't use `da->print(stream, req)`?
I recall some problems from not passing `req` to `DeclAttribute::printImpl` (namely the original function can't be inferred).

Reply:
Fixed.